### DISCUSSION TITLE
Nuxt 3 supabase error

### DISCUSSION THREAD

**Discussion by adityar15** [nuxt/nuxt#17352] [Questions]
I tried to include firebase with Nuxt 3, worked like a charm. I tried to integrate supabase with the same project, it started to throw the error of esm.

```
(node:18144) Warning: To load an ES module, set "type": "module" in the package.json or use the .mjs extension.
(Use `node --trace-warnings ...` to show where the warning was created)

 ERROR  [worker] [vite dev] Error loading external "L:/VueJS/projecttest/node_modules/@supabase/supabase-js/dist/module/lib/version.js".

  at /L:/VueJS/projecttest/.nuxt/dist/server/server.mjs:3026:300
  at async __instantiateModule__ (/L:/VueJS/projecttest/.nuxt/dist/server/server.mjs:7427:3)

```

When I put '@supabase/supabase-js' in transpile it still doesn't work. 
When I put '@supabase' in transpile it throws error for vue-router.
Aliasing doesn't work as well.

**Comment by adityar15** [comment]
Ok, that's very weird ... when I disable ssr through nuxt.config.ts it works fine.

**Reply by wohui** [reply]
it works fine without ssr? why ?

**Comment by psto** [comment]
I had the same conflict when importing supbase with SSR enabled. As you've noticed, disabling SSR "fixes" the issue.

A second workaround I found is using express as serverMiddleware and make an API request there:

```js
import express from 'express'
import { supabase } from '../plugins/supabase'
const app = express()

app.use(express.json())

app.post('/login', async (req, res) => {
  const { user, error } = await supabase.auth.signin(req.body)
  res.send({
    user, error
  })
})

export default app
```

In `nuxt.config.ts` add:

```js
serverMiddleware: [
  { path: '/api', handler: '~/server-middleware/index.js' },
]
```

Then we can make an axios request from the login page: `axios.post('api/login', payload).then...`

**Reply by adityar15** [reply]
Thanks for the reply. Yeah, I figured that out about the express api... hopefully it is addressed in the stable version of Nuxt as there must be many libraries who are doing export as Supabase.

**Reply by hassanshebbani** [reply]
issue still persists till this day.. any workaround other than the express thingy?
